### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by josh146**
Updates the QML readme to be more visually appealing, and to include more links back to the QML website. This should make the demonstration webpage more discoverable for users who find the repository via GitHub, as well as better articulate what this repo is.

In addition:

* Adds a 'contributions welcome' badge
* All screenshots are hyperlinks to their website location

The rendered readme can be viewed here: https://github.com/PennyLaneAI/qml/tree/update-readme

Screenshot:

### CODE REVIEW

**Review by co9olguy — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -18,6 +36,13 @@ as well as Python code demos. Both can be accessed at
...
+<a href="https://pennylane.ai/qml/demonstations.html">
+<img src="https://raw.githubusercontent.com/PennyLaneAI/qml/update-readme/_static/readme_panel3.png" width="900px">
+</a>
+
+Explore these materials on our website: https://pennylane.ai/qml. All tutorials are fully executable,
+and can be downloaded as Jupyter notebooks and Python scripts.
+
 ## Contributing
 
 You can contribute by submitting a demo implementing a recent
```
Comment (thisac) [inline README.md:48]: There's a line-break on line 60 that shouldn't be there. After `If you would like to write a demo...`.

File: README.md [hunk 2]
```diff
@@ -105,7 +130,7 @@ there are a couple of guidelines to keep in mind.
   be set for reproducibility.
```
Comment (thisac) [inline README.md:130]: On line 127 above, punctuation is missing after `Any included text should be minimal and legible`.

**Review by thisac — APPROVED** [review]
Thanks @josh146. This looks great! Just found two very minor things (that seemed to be there before).

File: README.md (hunk 1 above)
Comment (josh146) [reply README.md:48]: good catch!

**Comment by josh146** [comment]
Thanks @thisac and @co9olguy. I've updated the PR with the suggested changes, and also added in two new bullet points:

* The difference between the `master` and `dev` branches as per @ixfoduap's suggestion

* How to build only a single tutorial/demo

File: README.md [hunk 3]
```diff
@@ -45,7 +45,7 @@ and can be downloaded as Jupyter notebooks and Python scripts.
 
 ## Contributing
 
-You can contribute by submitting a demo implementing a recent
+You can contribute by submitting a demo via pull request implementing a recent
```
Comment (thisac) [inline README.md:48]: :100:

File: README.md [hunk 4]
```diff
@@ -102,6 +101,10 @@ quantum machine learning paper/result.
 
   and you are ready to submit a pull request!
 
+If your demo uses the latest release of PennyLane, simply make your PR against the
+`master` branch. If you instead require the cutting-edge development versions of
+PennyLane or any relevant plugins, make your PR against the `dev` branch instead.
+
```
Comment (thisac) [inline README.md:107]: Great addition!

Comment (ixfoduap) [reply README.md:107]: Awesome! Thanks for including this!

**Pull request merged**